### PR TITLE
7 jooq

### LINKED ISSUE #5: Evaluate Hikari DBPool
https://brettwooldridge.github.io/HikariCP/

### LINKED ISSUE #7: Evaluate JOOQ for database access
https://www.jooq.org/

### PR DESCRIPTION
**Pull request by Adeynack**
fix #7 
fix #5 

Onboard the `JOOQ` library for generating type-safe DSL based on latest state of the database.
Onboard the `Hikari` connection pool at the same time (closing #5).

### CODE REVIEW

**Comment by Adeynack** [comment]
@northfury : I would like you to have an eye on this since you were the inspiration for this PR.

File: src/main/kotlin/com/github/adeynack/finances/service/repository/BookRepository.kt [hunk 1]
```diff
@@ -1,42 +1,24 @@
...
+        .fetch(::toBook)
 
-    fun findAll(): List<Book> = ps("""
-        select
-          id,
-          name,
-          owner_id
-        from books
-    """).executeQuery().mapAndClose(toBook)
+    fun findById(id: UUID): List<Book> = databaseService.dslContext
```
Comment (NorthFury) [inline src/main/kotlin/com/github/adeynack/finances/service/repository/BookRepository.kt]: I would return `Book?`.

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -18,10 +18,13 @@ finances {
   database {
     dataSource {
       driver = "org.postgresql.Driver"
-      connectionString = "jdbc:postgresql://localhost:5432/postgres"
+      connectionString = "jdbc:postgresql://localhost:5432/finances-dev"
       username = "postgres"
       password = "postgres"
-      schema = "finances-dev"
+      schema = "finances" // has to stay the same accross configurations because JOOQ generated code has it hard-coded.
```
Comment (NorthFury) [inline src/main/resources/application.conf]: I didn't know this. I'll have to look into it as well.

Comment (NorthFury) [reply src/main/resources/application.conf]: Instead of using a different schema for dev, I would use a dev DB, which in turn is configured in Hikari.

Comment (Adeynack) [reply src/main/resources/application.conf]: @NorthFury : This is exactly what I am doing here. It was not possible to use different schemas, so I am simply using different databases now. Hence this comment.

**Pull request merged**